### ISSUE TITLE
[JENKINS-23787] EC2-plugin not spooling up stopped nodes - "still in the queue ... all nodes of label ... are offline"

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/ec2-plugin#1332]
<p>The Jenkins EC2 plugin no longer launches stopped nodes. Unfortunately I'm not sure exactly when it stopped working - I wasn't sure that was the issue until later, due to unrelated issues caused by too many nodes spawning and having to be killed.</p>

<p>If I use Manage Jenkins -> Manage Nodes to start a stopped EC2 node that a build is waiting on manually, the build proceeds.</p>

<p>Builds succeed when the EC2 plugin spawns a new node for the first time. It's only a problem if the node is stopped for idleness - the plugin doesn't seem to restart it.</p>

<p>Builds get stuck with output like:</p>

<pre>
Triggering bdr_linux ? x64,debian7
Triggering bdr_linux ? x86,amazonlinux201209
Triggering bdr_linux ? x86,debian6
Triggering bdr_linux ? x64,amazonlinux201209
Configuration bdr_linux ? x86,amazonlinux201209 is still in the queue: Amazon Linux 2012.09 EBS 32-bit  (i-b848fbfa) is offline
Configuration bdr_linux ? x86,amazonlinux201209 is still in the queue: All nodes of label ?amazonlinux201209&&x86? are offline
</pre>

<p>where there's at least one node with that label stopped, ready to start and use.</p>

<p>There's no sign that any attempt is made to start the node.</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ringerc.jpg" title="ringerc's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ringerc">ringerc</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-23787" target="_blank">EC2-plugin not spooling up stopped nodes - "still in the queue ... all nodes of label ... are offline"</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/francisu.jpg" title="francisu's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=francisu">francisu</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>component(s)</b>: ec2-plugin
<li><b>label(s)</b>: demand-launch, ec2, regression
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2015-10-25T04:45:00+00:00
<li><b>votes</b>: 2
<li><b>watchers</b>: 8
<li><b>imported</b>: 2025-12-06
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>The Jenkins EC2 plugin no longer launches stopped nodes. Unfortunately I'm not sure exactly when it stopped working - I wasn't sure that was the issue until later, due to unrelated issues caused by too many nodes spawning and having to be killed.</p>

<p>If I use Manage Jenkins -&gt; Manage Nodes to start a stopped EC2 node that a build is waiting on manually, the build proceeds.</p>

<p>Builds succeed when the EC2 plugin spawns a new node for the first time. It's only a problem if the node is stopped for idleness - the plugin doesn't seem to restart it.</p>

<p>Builds get stuck with output like:</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">Triggering bdr_linux ? x64,debian7
Triggering bdr_linux ? x86,amazonlinux201209
Triggering bdr_linux ? x86,debian6
Triggering bdr_linux ? x64,amazonlinux201209
Configuration bdr_linux ? x86,amazonlinux201209 is still in the queue: Amazon Linux 2012.09 EBS 32-bit  (i-b848fbfa) is offline
Configuration bdr_linux ? x86,amazonlinux201209 is still in the queue: All nodes of label ?amazonlinux201209&amp;&amp;x86? are offline
</pre>
</div></div>

<p>where there's at least one node with that label stopped, ready to start and use.</p>

<p>There's no sign that any attempt is made to start the node.</p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>Jenkins 1.572, EC2 plugin 1.21, Node Iterator API Plugin 1.5</code></li></ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ringerc.jpg" title="ringerc's avatar" /> ringerc:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/23787?focusedId=205730&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-205730">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-23787?focusedId=205730&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-205730">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>After updating to 1.23 I instead get the behaviour in <a href="https://issue-redirect.jenkins.io/issue/23788" title="EC2 plugin regression: fails to launch nodes with &quot;syntax error near unexpected token `newline'&quot;" class="issue-link" data-issue-key="JENKINS-23788"><del>JENKINS-23788</del></a> . Manual node launch no longer works.</p>
</pre>
</details>

</ul>
</details>

<p>After updating to 1.23 I instead get the behaviour in <a href="https://issue-redirect.jenkins.io/issue/23788" title="EC2 plugin regression: fails to launch nodes with "syntax error near unexpected token `newline'"" class="issue-link" data-issue-key="JENKINS-23788"><del>JENKINS-23788</del></a> . Manual node launch no longer works.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/acoulton.jpg" title="acoulton's avatar" /> acoulton:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/23787?focusedId=205832&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-205832">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-23787?focusedId=205832&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-205832">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I think I am seeing a similar issue with EC2 1.23 and Node Iterator 1.5.</p>

<p>Every time I start a build, Jenkins launches a new slave rather than restarting one of the stopped instances. It is successfully stopping the instance when it hits the idle time.</p>

<p>During idle, the log shows lots of this (the _check, idleTimeout, stop entries appear once for every instance currently registered):</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">Jul 16, 2014 12:40:47 PM hudson.model.AsyncPeriodicWork$1 run
INFO: Started EC2 alive slaves monitor
Jul 16, 2014 12:40:48 PM hudson.model.AsyncPeriodicWork$1 run
INFO: Finished EC2 alive slaves monitor. 1172 ms
Jul 16, 2014 12:41:53 PM hudson.plugins.ec2.EC2RetentionStrategy _check
INFO: Idle timeout: edifestivalsapi build slave (i-ce32a08c)
Jul 16, 2014 12:41:53 PM hudson.plugins.ec2.EC2AbstractSlave idleTimeout
INFO: EC2 instance idle time expired: i-ce32a08c
Jul 16, 2014 12:41:53 PM hudson.plugins.ec2.EC2AbstractSlave stop
INFO: EC2 instance stopped: i-ce32a08c
</pre>
</div></div>

<p>Then a build is triggered and the idle timeout checks run again (again, one set of entries for every instance):</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">Jul 16, 2014 12:44:23 PM com.cloudbees.jenkins.GitHubPushTrigger$1 run
INFO: SCM changes detected in edifestivalsapi-master. Triggering  #36
Jul 16, 2014 12:45:53 PM hudson.plugins.ec2.EC2RetentionStrategy _check
INFO: Idle timeout: edifestivalsapi build slave (i-ce32a08c)
Jul 16, 2014 12:45:53 PM hudson.plugins.ec2.EC2AbstractSlave idleTimeout
INFO: EC2 instance idle time expired: i-ce32a08c
Jul 16, 2014 12:45:53 PM hudson.plugins.ec2.EC2AbstractSlave stop
INFO: EC2 instance stopped: i-ce32a08c
</pre>
</div></div>

<p>And then the plugin starts to provision a new instance - apparently without any attempt to restart a stopped slave.</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">Jul 16, 2014 12:46:33 PM hudson.plugins.ec2.EC2Cloud provision
INFO: Excess workload after pending Spot instances: 1
Jul 16, 2014 12:46:33 PM hudson.plugins.ec2.EC2Cloud addProvisionedSlave
INFO: Provisioning <span class="code-keyword">for</span> AMI ami-57ea3d20; Estimated number of total slaves: 0; Estimated number of slaves <span class="code-keyword">for</span> ami ami-57ea3d20: 0
Launching ami-57ea3d20 <span class="code-keyword">for</span> template edifestivalsapi build slave
Jul 16, 2014 12:46:33 PM hudson.slaves.NodeProvisioner update
INFO: Started provisioning edifestivalsapi build slave (ami-57ea3d20) from ec2-eu-west-1 with 1 executors. Remaining excess workload:0.0
Looking <span class="code-keyword">for</span> existing instances: {InstanceIds: [],Filters: [{Name: image-id,Values: [ami-57ea3d20]}, {Name: group-name,Values: [jenkins-build-slave]}, {Name: key-name,Values: [build-slave]}, {Name: instance-type,Values: [t1.micro]}, {Name: tag:Name,Values: [edifestivalsapi-build-slave]}, {Name: tag:Project,Values: [edifestivalsapi]}, {Name: instance-state-name,Values: [stopped, stopping]}],}
No existing instance found - created: {InstanceId: i-eb35a8a9,ImageId: ami-57ea3d20,State: {Code: 0,Name: pending},"**REDACTED**}
</pre>
</div></div>

<p>Then another block of the idle timeout checks while the instance is launched, and then this:</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">Jul 16, 2014 12:47:44 PM hudson.slaves.NodeProvisioner update
INFO: edifestivalsapi build slave (ami-57ea3d20) provisioningE successfully completed. We have now 8 computer(s)
Jul 16, 2014 12:47:47 PM hudson.node_monitors.ResponseTimeMonitor$1 monitor
WARNING: Making edifestivalsapi build slave (i-ce32a08c) offline because it’s not responding
</pre>
</div></div>

<p>The UI shows all the slaves that were launched for previous jobs but shows them as offline with "Time out for last 5 try". When I manually start the instance (by clicking onto the slave page and clicking "Launch slave agent") I see that the stopped instance is restarted and comes online as expected.</p>

<p>So my hunch is that that Jenkins somehow isn't detecting that it has a stopped instance for the given AMI?</p>
</pre>
</details>

</ul>
</details>

<p>I think I am seeing a similar issue with EC2 1.23 and Node Iterator 1.5.</p>

<p>Every time I start a build, Jenkins launches a new slave rather than restarting one of the stopped instances. It is successfully stopping the instance when it hits the idle time.</p>

<p>During idle, the log shows lots of this (the _check, idleTimeout, stop entries appear once for every instance currently registered):</p>


<pre>
Jul 16, 2014 12:40:47 PM hudson.model.AsyncPeriodicWork$1 run
INFO: Started EC2 alive slaves monitor
Jul 16, 2014 12:40:48 PM hudson.model.AsyncPeriodicWork$1 run
INFO: Finished EC2 alive slaves monitor. 1172 ms
Jul 16, 2014 12:41:53 PM hudson.plugins.ec2.EC2RetentionStrategy _check
INFO: Idle timeout: edifestivalsapi build slave (i-ce32a08c)
Jul 16, 2014 12:41:53 PM hudson.plugins.ec2.EC2AbstractSlave idleTimeout
INFO: EC2 instance idle time expired: i-ce32a08c
Jul 16, 2014 12:41:53 PM hudson.plugins.ec2.EC2AbstractSlave stop
INFO: EC2 instance stopped: i-ce32a08c
</pre>

<p>Then a build is triggered and the idle timeout checks run again (again, one set of entries for every instance):</p>


<pre>
Jul 16, 2014 12:44:23 PM com.cloudbees.jenkins.GitHubPushTrigger$1 run
INFO: SCM changes detected in edifestivalsapi-master. Triggering  #36
Jul 16, 2014 12:45:53 PM hudson.plugins.ec2.EC2RetentionStrategy _check
INFO: Idle timeout: edifestivalsapi build slave (i-ce32a08c)
Jul 16, 2014 12:45:53 PM hudson.plugins.ec2.EC2AbstractSlave idleTimeout
INFO: EC2 instance idle time expired: i-ce32a08c
Jul 16, 2014 12:45:53 PM hudson.plugins.ec2.EC2AbstractSlave stop
INFO: EC2 instance stopped: i-ce32a08c
</pre>

<p>And then the plugin starts to provision a new instance - apparently without any attempt to restart a stopped slave.</p>


<pre>
Jul 16, 2014 12:46:33 PM hudson.plugins.ec2.EC2Cloud provision
INFO: Excess workload after pending Spot instances: 1
Jul 16, 2014 12:46:33 PM hudson.plugins.ec2.EC2Cloud addProvisionedSlave
INFO: Provisioning <span class="code-keyword">for</span> AMI ami-57ea3d20; Estimated number of total slaves: 0; Estimated number of slaves <span class="code-keyword">for</span> ami ami-57ea3d20: 0
Launching ami-57ea3d20 <span class="code-keyword">for</span> template edifestivalsapi build slave
Jul 16, 2014 12:46:33 PM hudson.slaves.NodeProvisioner update
INFO: Started provisioning edifestivalsapi build slave (ami-57ea3d20) from ec2-eu-west-1 with 1 executors. Remaining excess workload:0.0
Looking <span class="code-keyword">for</span> existing instances: {InstanceIds: [],Filters: [{Name: image-id,Values: [ami-57ea3d20]}, {Name: group-name,Values: [jenkins-build-slave]}, {Name: key-name,Values: [build-slave]}, {Name: instance-type,Values: [t1.micro]}, {Name: tag:Name,Values: [edifestivalsapi-build-slave]}, {Name: tag:Project,Values: [edifestivalsapi]}, {Name: instance-state-name,Values: [stopped, stopping]}],}
No existing instance found - created: {InstanceId: i-eb35a8a9,ImageId: ami-57ea3d20,State: {Code: 0,Name: pending},"**REDACTED**}
</pre>

<p>Then another block of the idle timeout checks while the instance is launched, and then this:</p>


<pre>
Jul 16, 2014 12:47:44 PM hudson.slaves.NodeProvisioner update
INFO: edifestivalsapi build slave (ami-57ea3d20) provisioningE successfully completed. We have now 8 computer(s)
Jul 16, 2014 12:47:47 PM hudson.node_monitors.ResponseTimeMonitor$1 monitor
WARNING: Making edifestivalsapi build slave (i-ce32a08c) offline because it’s not responding
</pre>

<p>The UI shows all the slaves that were launched for previous jobs but shows them as offline with "Time out for last 5 try". When I manually start the instance (by clicking onto the slave page and clicking "Launch slave agent") I see that the stopped instance is restarted and comes online as expected.</p>

<p>So my hunch is that that Jenkins somehow isn't detecting that it has a stopped instance for the given AMI?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ringerc.jpg" title="ringerc's avatar" /> ringerc:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/23787?focusedId=205913&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-205913">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-23787?focusedId=205913&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-205913">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<blockquote>
<p>So my hunch is that that Jenkins somehow isn't detecting that it has a stopped instance for the given AMI?</p></blockquote>

<p>That used to happen due to a bug when label support was added, but it's fixed in 1.21 IIRC.</p>

<p>I'm not seeing the same behaviour - rather, it's just waiting indefinitely for the stopped node to start.</p>

<p>It might be a good idea to file a separate issue for what you're discussing here, then comment to mention the issue number here in case they prove to be related.</p>
</pre>
</details>

</ul>
</details>

<blockquote>
<p>So my hunch is that that Jenkins somehow isn't detecting that it has a stopped instance for the given AMI?</p></blockquote>

<p>That used to happen due to a bug when label support was added, but it's fixed in 1.21 IIRC.</p>

<p>I'm not seeing the same behaviour - rather, it's just waiting indefinitely for the stopped node to start.</p>

<p>It might be a good idea to file a separate issue for what you're discussing here, then comment to mention the issue number here in case they prove to be related.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/acoulton.jpg" title="acoulton's avatar" /> acoulton:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/23787?focusedId=205934&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-205934">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-23787?focusedId=205934&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-205934">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Thanks Craig, I've filed as a separate issue at <a href="https://issues.jenkins-ci.org/browse/JENKINS-23850" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-23850</a>. Do you have an instance cap? It's possible we're seeing the same thing if your Jenkins has hit the instance cap and therefore can't start a new node (so the build stalls), while mine is uncapped so just goes ahead and makes a new one.</p>
</pre>
</details>

</ul>
</details>

<p>Thanks Craig, I've filed as a separate issue at <a href="https://issues.jenkins-ci.org/browse/JENKINS-23850" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-23850</a>. Do you have an instance cap? It's possible we're seeing the same thing if your Jenkins has hit the instance cap and therefore can't start a new node (so the build stalls), while mine is uncapped so just goes ahead and makes a new one.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/acoulton.jpg" title="acoulton's avatar" /> acoulton:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/23787?focusedId=206007&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-206007">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-23787?focusedId=206007&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-206007">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>OK, does sound like yours is a different issue then.</p>
</pre>
</details>

</ul>
</details>

<p>OK, does sound like yours is a different issue then.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ringerc.jpg" title="ringerc's avatar" /> ringerc:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/23787?focusedId=206013&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-206013">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-23787?focusedId=206013&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-206013">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>(Removed; this should've been a comment on <a href="https://issue-redirect.jenkins.io/issue/23792" title="PATCH: EC2 plugin idles-down nodes that are still launching" class="issue-link" data-issue-key="JENKINS-23792"><del>JENKINS-23792</del></a> instead.)</p>
</pre>
</details>

</ul>
</details>

<p>(Removed; this should've been a comment on <a href="https://issue-redirect.jenkins.io/issue/23792" title="PATCH: EC2 plugin idles-down nodes that are still launching" class="issue-link" data-issue-key="JENKINS-23792"><del>JENKINS-23792</del></a> instead.)</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/francisu.jpg" title="francisu's avatar" /> francisu:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/23787?focusedId=235676&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-235676">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-23787?focusedId=235676&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-235676">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I have just made a fix for this in unreleased 1.30, if you can get the SNAPSHOT version of the ec2-plugin, hopefully this is fixed. Please give it and try and let me know.</p>
</pre>
</details>

</ul>
</details>

<p>I have just made a fix for this in unreleased 1.30, if you can get the SNAPSHOT version of the ec2-plugin, hopefully this is fixed. Please give it and try and let me know.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/francisu.jpg" title="francisu's avatar" /> francisu:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/23787?focusedId=235688&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-235688">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-23787?focusedId=235688&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-235688">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Thanks for letting me know. I will close this one and reopen my duplicated report, as that has to do with hooking up to EC2 instances that are actually running (even though the corresponding Jenkins slave is offline).</p>
</pre>
</details>

</ul>
</details>

<p>Thanks for letting me know. I will close this one and reopen my duplicated report, as that has to do with hooking up to EC2 instances that are actually running (even though the corresponding Jenkins slave is offline).</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>fengxx:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/23787?focusedId=236554&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-236554">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-23787?focusedId=236554&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-236554">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I think the commit <a href="https://github.com/jenkinsci/ec2-plugin/commit/c9d69a5da8c9be094701d4c191ba7b1d06c200c9" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/ec2-plugin/commit/c9d69a5da8c9be094701d4c191ba7b1d06c200c9</a> breaks plugin,  user can not launch multiple instances for same AMI since DescribeInstancesRequest returns all instances instead of stopped instances, and no instances can be launched if there is any running</p>

<p> below line was removed
diFilters.add(new Filter("instance-state-name").withValues(InstanceStateName.Stopped.toString(), InstanceStateName.Stopping.toString()));</p>
</pre>
</details>

</ul>
</details>

<p>I think the commit <a href="https://github.com/jenkinsci/ec2-plugin/commit/c9d69a5da8c9be094701d4c191ba7b1d06c200c9" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/ec2-plugin/commit/c9d69a5da8c9be094701d4c191ba7b1d06c200c9</a> breaks plugin,  user can not launch multiple instances for same AMI since DescribeInstancesRequest returns all instances instead of stopped instances, and no instances can be launched if there is any running</p>

<p> below line was removed<br/>
diFilters.add(new Filter("instance-state-name").withValues(InstanceStateName.Stopped.toString(), InstanceStateName.Stopping.toString()));</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/francisu.jpg" title="francisu's avatar" /> francisu:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/23787?focusedId=239930&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-239930">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-23787?focusedId=239930&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-239930">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Should be released in 1.30</p>
</pre>
</details>

</ul>
</details>

<p>Should be released in 1.30</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/francisu.jpg" title="francisu's avatar" /> francisu:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/23787?focusedId=245317&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-245317">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-23787?focusedId=245317&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-245317">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Fixed a problem with this of it not respecting the instance caps for on-demand nodes.</p>
</pre>
</details>

</ul>
</details>

<p>Fixed a problem with this of it not respecting the instance caps for on-demand nodes.</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `relates_to` from Jira: <a href="https://github.com/jenkinsci/ec2-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-23792%22">JENKINS-23792</a>]</i>
